### PR TITLE
Update DEVGUIDE.md

### PR DESCRIPTION
**Pull request by smoothdeveloper**
Add a section about updating FCS surface area baselines specifically.

Credit to @auduchinok

### CODE REVIEW

File: DEVGUIDE.md [hunk 1]
```diff
@@ -196,6 +196,17 @@ or
 > **Note**
 > Please note, that by default, **Release** version of IL baseline tests will be running in CI, so when updating baseline (.bsl) files, make sure to add `-c Release` flag to the build command.
 
+
+### Updating FCS surface area baselines
+
+```bash
+$env:TEST_UPDATE_BSL=1
```
Comment (vzarytovskii) [inline DEVGUIDE.md:203]: This is not needed for surface area, only test baselines (unless something changed lately, and if it does, it needs to be reverted to not require that flag).

File: DEVGUIDE.md [hunk 2]
```diff
@@ -196,6 +196,17 @@ or
...
+dotnet test tests/FSharp.Compiler.Service.Tests/FSharp.Compiler.Service.Tests.fsproj --filter "SurfaceAreaTest" /p:BUILDING_USING_DOTNET=true
```
Comment (vzarytovskii) [inline DEVGUIDE.md:204]: This won't work if building with build scripts. The dotnet building flag is not officially supported with arcade builds, and will yield cryptic errors.

This should instead explain how to update surface area with build scripts and separately with dotnet test, same as it is done with builds.

File: DEVGUIDE.md [hunk 3]
```diff
@@ -196,6 +196,17 @@ or
...
+dotnet test tests/FSharp.Compiler.Service.Tests/FSharp.Compiler.Service.Tests.fsproj --filter "SurfaceAreaTest" /p:BUILDING_USING_DOTNET=true
+dotnet test tests/FSharp.Compiler.Service.Tests/FSharp.Compiler.Service.Tests.fsproj --filter "SurfaceAreaTest" -c Release /p:BUILDING_USING_DOTNET=true
+dotnet test tests/FSharp.Compiler.Service.Tests/FSharp.Compiler.Service.Tests.fsproj --filter "SurfaceAreaTest" -c Release /p:BUILDING_USING_DOTNET=true
```
Comment (vzarytovskii) [inline DEVGUIDE.md:207]: Why is it here twice?

File: DEVGUIDE.md [hunk 4]
```diff
@@ -196,6 +196,17 @@ or
 > **Note**
 > Please note, that by default, **Release** version of IL baseline tests will be running in CI, so when updating baseline (.bsl) files, make sure to add `-c Release` flag to the build command.
 
+
+### Updating FCS surface area baselines
+
+```bash
+$env:TEST_UPDATE_BSL=1
+dotnet test tests/FSharp.Compiler.Service.Tests/FSharp.Compiler.Service.Tests.fsproj --filter "SurfaceAreaTest" /p:BUILDING_USING_DOTNET=true
+dotnet test tests/FSharp.Compiler.Service.Tests/FSharp.Compiler.Service.Tests.fsproj --filter "SurfaceAreaTest" /p:BUILDING_USING_DOTNET=true
```
Comment (vzarytovskii) [inline DEVGUIDE.md:205]: Only release run is probably needed. Don't think we care about debug surface area. If we do, debug test shall be removed.

**Review by vzarytovskii — CHANGES_REQUESTED** [review]

File: DEVGUIDE.md (hunk 3 above)
Comment (smoothdeveloper) [reply DEVGUIDE.md:207]: First pass, the test will fail (and update the baseline), second pass, it is green.

Comment (vzarytovskii) [reply DEVGUIDE.md:207]: But second isn't necessarily needed probably, should mention that and say that second one is optional.

File: DEVGUIDE.md (hunk 1 above)
Comment (nojaf) [reply DEVGUIDE.md:203]: You sure?
https://github.com/dotnet/fsharp/blob/4c60c63c3a0fb3c7507ced9f2124966e67e7d71b/tests/FSharp.Compiler.Service.Tests/SurfaceArea.fs#L9-L18

Comment (vzarytovskii) [reply DEVGUIDE.md:203]: It used to not rely. That was the whole point of it, so when you run tests, you update it and it was part of PR and reflected there.

Comment (vzarytovskii) [reply DEVGUIDE.md:203]: It was always updating it, I am not sure why would we make it not to:

https://github.com/dotnet/fsharp/commit/4c60c63c3a0fb3c7507ced9f2124966e67e7d71b

**Review by abonie — APPROVED** [review]

**Pull request merged**